data-processing-lib/ray/src/data_processing/transform/binary_transform.py

```diff
@@ -10,10 +10,15 @@
 # limitations under the License.
 ################################################################################
 
-from typing import Any
+from typing import Any, TypeVar
 
+from data_processing.transform.abstract_transform import AbstractTransform
 
-class AbstractBinaryTransform:
+
+DATA = TypeVar("DATA")
+
+
```

Comment:
But we need to enable passing DATA to the super class as a generic type.  See line 21 below.

Reply:
That is the complexity I am talking about. Why?

Reply:
from comment below
`This comes in because of the AbstractTransformLauncher -> runtime config -> transform config -> AbstractTransform. Spark code is using the AbstractTransformLauncher and therefore we have a common AbstractTransform across all runtimes/launchers.`

Reply:
And the same question. How is this helping us?